geopandas/tools/overlay.py

```diff
@@ -176,3 +180,204 @@ def overlay(df1, df2, how, use_sindex=True):
+    df1 : GeoDataFrame with MultiPolygon or Polygon geometry column
+    df2 : GeoDataFrame with MultiPolygon or Polygon geometry column
+    how : string
+        Method of spatial overlay: 'intersection', 'union',
+        'identity', 'symmetric_difference' or 'difference'.
+    reproject : boolean, default True
+        If GeoDataFrames do not have same projection, reproject
+        df2 to same projection of df1 before performing overlay
+
+    Returns
+    -------
+    df : GeoDataFrame
+        GeoDataFrame with new set of polygons and attributes
+        resulting from the overlay
+
+    """
+    if use_sindex is not None:
+        warnings.warn("'use_sindex' is deprecated. The overlay operation "
+                      "always requires a spatial index (rtree).",
+                      DeprecationWarning, stacklevel=2)
+
+    # Allowed operations
+    allowed_hows = [
+        'intersection',
+        'union',
+        'identity',
+        'symmetric_difference',
+        'difference',  # aka erase
+    ]
+    # Error Messages
+    if how not in allowed_hows:
+        raise ValueError("`how` was '{0}' but is expected to be "
+                         "in %s".format(how, allowed_hows))
+
+    if isinstance(df1, GeoSeries) or isinstance(df2, GeoSeries):
+        raise NotImplementedError("overlay currently only implemented for "
+                                  "GeoDataFrames")
+
+    if (not df1.geom_type.isin(['Polygon', 'MultiPolygon']).all()
+            or not df2.geom_type.isin(['Polygon', 'MultiPolygon']).all()):
+        raise TypeError("overlay only takes GeoDataFrames with (multi)polygon "
+                        " geometries.")
+
+    # Computations
+    df1 = df1.copy()
+    df2 = df2.copy()
+    df1[df1._geometry_column_name] = df1.geometry.buffer(0)
+    df2[df2._geometry_column_name] = df2.geometry.buffer(0)
+    if df1.crs != df2.crs and reproject:
+        warnings.warn("Data has different projections. Converted data to "
+                      "projection of first GeoPandas GeoDataFrame",
+                      UserWarning)
+        df2.to_crs(crs=df1.crs, inplace=True)
+    if how == 'intersection':
+        result = overlay_intersection(df1, df2)
+        result.drop(['__idx1', '__idx2'], axis=1, inplace=True)
+        return result
+    elif how == 'difference':
+        return overlay_difference(df1, df2)
+    elif how == 'symmetric_difference':
+        result = overlay_symmetric_diff(df1, df2)
+        result.drop(['__idx1', '__idx2'], axis=1, inplace=True)
+        return result
+    elif how == 'union':
+        result = overlay_union(df1, df2)
+        result.drop(['__idx1', '__idx2'], axis=1, inplace=True)
+        return result
+    elif how == 'identity':
+        dfunion = overlay_union(df1, df2)
+        result = dfunion[dfunion['__idx1'].notna()].copy()
+        result.reset_index(drop=True, inplace=True)
+        result.drop(['__idx1', '__idx2'], axis=1, inplace=True)
+        return result
```

Comment:
Both of the final two lines here appear to be called in all the above if operations. Could they just be placed once, at the bottom of this method, after the if checks?

Such that the parent method just ends with:
```result.drop(['__idx1', '__idx2'], axis=1, inplace=True)
return result```

Reply:
yep, good idea